Mathlib/Topology/PartialHomeomorph.lean

```diff
@@ -1393,7 +1393,7 @@ end OpenEmbedding
 /- inclusion of an open set in a topological space -/
 namespace TopologicalSpace.Opens
 
-variable (s : Opens α) [Nonempty s]
+variable (s : Opens α) (hs : Nonempty s)
```

Comment:
You are using the Type version `Nonempty`. Wouldn't it be more natural to use the set version `Set.Nonempty`, as you are dealing with a subset?

Reply:
This was a conscious choice. When I used `Set.Nonempty`, the proof obligation for that argument when provided with an element of the set is somehow longer. Instead of `⟨x⟩`, you'd have to write `⟨x, x.prop⟩`. Unless I've missed an shorter spelling, I would find the type `Nonempty` easier to use.

Reply:
ok, that's perfect if it's a deliberate design choice. Maybe add a comment explaining this?